### PR TITLE
[FL-3284] Fix reading Mifare Classic cards with unusual access conditions and fix emulation of unknown keys

### PR DESCRIPTION
**Pull request by Astrrra**
# What's new

- Fixes one of the problems in #2577 (reading 30 keys instead of the actual 31 known keys)
- When emulating Mifare Classic with unknown keys the Flipper does not assume that the unknown keys are 00000000000 anymore.

# Verification 

- Include the non-standard keys in the user dict and read the dump from #2577, make sure that 31 keys are found.
- Emulate a card with unknown keys, try authenticating to the unknown sectors with 00000000000 as a key, make sure that the authentication fails.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by xMasterX** [comment]
Hi, `Detect reader` will not collect any nonces with changes introduced in this PR
Something is not right with changes in this file, can't say what exactly, haven't checked further
`lib/nfc/protocols/mifare_classic.c`

**Comment by Astrrra** [comment]
@xMasterX nice catch, thanks! Will fix soon

File: lib/nfc/nfc_worker.c [hunk 1]
```diff
@@ -696,8 +697,14 @@ void nfc_worker_mf_classic_dict_attack(NfcWorker* nfc_worker) {
...
                     // If the key A is marked as found and matches the searching key, invalidate it
+                    uint8_t found_key[6];
+                    memcpy(found_key, data->block[i].value, 6);
+
+                    uint8_t current_key[6];
+                    memcpy(current_key, &key, 6);
+
                     if(mf_classic_is_key_found(data, i, MfClassicKeyA) &&
-                       data->block[i].value[0] == key) {
+                       found_key == current_key) {
```
Comment (skotopes) [inline lib/nfc/nfc_worker.c]: comparing pointers instead of comparing content is clearly incorrect here

File: lib/nfc/nfc_worker.c [hunk 2]
```diff
@@ -707,16 +714,23 @@ void nfc_worker_mf_classic_dict_attack(NfcWorker* nfc_worker) {
...
                     // If the key B is marked as found and matches the searching key, invalidate it
+                    uint8_t found_key[6];
+                    memcpy(found_key, data->block[i].value + 10, 6);
+
+                    uint8_t current_key[6];
+                    memcpy(current_key, &key, 6);
+
                     if(mf_classic_is_key_found(data, i, MfClassicKeyB) &&
-                       data->block[i].value[10] == key) {
+                       found_key == current_key) {
```
Comment (skotopes) [inline lib/nfc/nfc_worker.c]: same

**Review by skotopes — CHANGES_REQUESTED** [review]

**Review by skotopes — APPROVED** [review]

**Pull request merged**